seirmo/tests/test_models.py

```diff
@@ -125,5 +128,135 @@ def test_simulate(self):
+        self.assertEqual(reduced_model._fixed_params_values[0], 0.5)
+        # test the value of alpha
+        self.assertEqual(reduced_model._fixed_params_values[4], 1)
+
+        # Test that the mask and values are None when all parameters are free
+        name_value_dict = {'S0': None, 'alpha': None}
+        reduced_model.fix_parameters(name_value_dict)
+        self.assertEqual(reduced_model._fixed_params_mask, None)
+        self.assertEqual(reduced_model._fixed_params_values, None)
+
+    def test_n_fixed_parameters(self):
+        # Test the number of fixed parameters is as expected
+        reduced_model = se.ReducedModel(se.SEIRModel())
+        # Test the case when self._fixed_params_mask is None:
+        self.assertEqual(reduced_model.n_fixed_parameters(), 0)
+        name_value_dict = {'S0': 0.5, 'alpha': 1}
+        reduced_model.fix_parameters(name_value_dict)
+        self.assertEqual(reduced_model.n_fixed_parameters(), 2)
+
+    def test_n_outputs(self):
+        # Test the number of outputs is as expected
+        reduced_model = se.ReducedModel(se.SEIRModel())
+        self.assertEqual(reduced_model.n_outputs(), 5)
+
+    def test_n_parameters(self):
+        # Test the number of (unfixed) parameters is as expected
+        reduced_model = se.ReducedModel(se.SEIRModel())
+
+        # The case when the mask is None
+        self.assertEqual(reduced_model.n_parameters(), 7)
+
+        name_value_dict = {'S0': 0.5, 'alpha': 1}
+        reduced_model.fix_parameters(name_value_dict)
+        self.assertEqual(reduced_model.n_parameters(), 5)
+
+    def test_output_names(self):
+        # Test the output names are as expected
+        reduced_model = se.ReducedModel(se.SEIRModel())
+        self.assertEqual(reduced_model.output_names(), [
+            'S', 'E', 'I', 'R', 'Incidence'
+        ])
+
+        reduced_model.set_outputs(['I', 'Incidence'])
+        self.assertEqual(reduced_model.output_names(), ['I', 'Incidence'])
+
+    def test_parameter_names(self):
+        # Test the parameter names are as expected
+        reduced_model = se.ReducedModel(se.SEIRModel())
+        name_value_dict = {'S0': 0.5, 'alpha': 1}
+        reduced_model.fix_parameters(name_value_dict)
+        self.assertEqual(
+            reduced_model.parameter_names(),
+            ['E0', 'I0', 'R0', 'beta', 'gamma'])
+
+    def test_set_outputs(self):
+        reduced_model = se.ReducedModel(se.SEIRModel())
+
+        # Check ValueError will be raised when some output names
+        # are not as required
+        with self.assertRaises(ValueError):
+            reduced_model.set_outputs(['incidence number'])
+
+        reduced_model.set_outputs(['I', 'Incidence'])
+        # Check the outputs names and number are as expected
+        self.assertEqual(reduced_model.output_names(), ['I', 'Incidence'])
+        self.assertEqual(reduced_model.n_outputs(), 2)
+
+    def test_simulate(self):
+        reduced_model = se.ReducedModel(se.SEIRModel())
```

Comment:
As mentioned above it's probably good to compare the simulation results of a reduced model and a normal model with the same parameters.